ambry-api/src/main/java/com.github.ambry/clustermap/ClusterMap.java

```diff
@@ -122,6 +122,15 @@
    */
   JSONObject getSnapshot();
 
+  /**
+   * Get new replica of certain partition that resides on given host.
```

Comment:
It's possible that one host has several new replicas but they should come from different partitions. In this method, neither argument `partitionIdStr` nor return value `ReplicaId` is collection, because the method is supposed to be called by state model where each partition get its new replica separately.  I will add some comments for this method.